src/makeConfig.ts

```diff
@@ -0,0 +1,69 @@
+import { Options, FetchContextTypes, OptionsMaybeURL, NoUrlOptions } from './types'
+import { isString, isObject, pullOutRequestInit } from "./utils"
+
+export const DefaultOptions: NoUrlOptions = {
+  onMount: false
+}
+
+function makeConfig(context: FetchContextTypes, urlOrOptions?: string | OptionsMaybeURL, optionsNoURLs?: NoUrlOptions): Options {
+  let options: Partial<Options> = {}
+  let requestInit: RequestInit = {}
+
+  if (isObject(urlOrOptions) && isObject(optionsNoURLs)) {
+    // TODO: I could not get the utils/invariant to work with the condition isObject(urlOrOptions) && isObject(optionsNoURLs)
+    throw new Error('You cannot have a 2nd parameter of useFetch when your first argument is a object config.')
+  }
+
+  const getUrl = (): string => {
+    if (isString(urlOrOptions)) {
+      return urlOrOptions
+    }
+
+    if (isObject(urlOrOptions) && !!urlOrOptions.url) {
+      return urlOrOptions.url;
+    }
+
+    if (isObject(urlOrOptions) && !!urlOrOptions.url) {
```

Comment:
isn't this a duplicate of line 22?

Reply:
yes sir, great spot.  3 less lines of code woohoo